cli/azd/pkg/tools/azcli/azcli.go

```diff
@@ -60,7 +60,7 @@ type AzCli interface {
 	Login(ctx context.Context, useDeviceCode bool, deviceCodeWriter io.Writer) error
 	LoginAcr(ctx context.Context, subscriptionId string, loginServer string) error
 	GetContainerRegistries(ctx context.Context, subscriptionId string) ([]*armcontainerregistry.Registry, error)
-	ListAccounts(ctx context.Context) ([]AzCliSubscriptionInfo, error)
+	ListAccounts(ctx context.Context) ([]*AzCliSubscriptionInfo, error)
```

Comment:
Did you have a specific reason for this? `slices` are reference types, so they can already be set to `nil`, you wouldn't need a pointer unless you're trying to avoid copies of the same subscription info object

Reply:
The slice isn't nil, just the items in the slice are pointers instead of values.  No reason to copy these objects around.